internal/util/observability/otel_propagation.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observability
+
+import (
+	"encoding/hex"
+	"encoding/json"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// TraceData represents OpenTelemetry tracing information that can be used to restore the info about parent span.
+type TraceData struct {
+	TraceID [16]byte `json:"traceID"`
+	SpanID  [8]byte  `json:"spanID"`
+}
+
+// MarshalJSON implements json.Marshaler interface.
+func (td *TraceData) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		TraceID string `json:"traceID"`
+		SpanID  string `json:"spanID"`
+	}{
+		TraceID: hex.EncodeToString(td.TraceID[:]),
+		SpanID:  hex.EncodeToString(td.SpanID[:]),
+	})
+}
+
+// UnmarshalJSON implements json.Unmarshaler interface.
+func (td *TraceData) UnmarshalJSON(data []byte) error {
+	var stringData struct {
+		TraceID string `json:"traceID"`
+		SpanID  string `json:"spanID"`
```

Comment:
I think there was an earlier comment about not duplicated format such as `json:"spanID"`, I think we can only define format once right?